libraries/lib-xml/XMLAttributeValueView.cpp

```diff
@@ -0,0 +1,415 @@
+
+      double tempValue = {};
+
+      const auto result = FromChars(mStringView.Data, end, tempValue);
+
+      if (result.ec == std::errc() && result.ptr == end)
+      {
+         value = tempValue;
+         return true;
+      }
+   }
+
+   return false;
+}
+
+bool XMLAttributeValueView::Get(std::string_view& value) const noexcept
+{
+   if (mType != Type::StringView)
+      return false;
+
+   value = std::string_view(mStringView.Data, mStringView.Length);
+
+   return true;
+}
+
+bool XMLAttributeValueView::GetBool(bool defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+int16_t XMLAttributeValueView::GetInt16(
+   short defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+uint16_t XMLAttributeValueView::GetUInt16(
+   unsigned short defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+int32_t
+XMLAttributeValueView::GetInt32(int32_t defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+uint32_t
+XMLAttributeValueView::GetUInt32(uint32_t defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+int64_t
+XMLAttributeValueView::GetInt64(int64_t defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+uint64_t
+XMLAttributeValueView::GetUInt64(uint64_t defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+float XMLAttributeValueView::GetFloat(
+   float defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+double
+XMLAttributeValueView::GetDouble(double defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+std::string_view XMLAttributeValueView::GetStringView(
+   std::string_view defaultValue /*= {}*/) const noexcept
+{
+   Get(defaultValue);
+   return defaultValue;
+}
+
+std::string XMLAttributeValueView::ToString() const
+{
+   switch (mType)
+   {
+   case XMLAttributeValueView::Type::Null:
+      return {};
+   case XMLAttributeValueView::Type::SignedInteger:
+      return std::to_string(mInteger);
+   case XMLAttributeValueView::Type::UnsignedInteger:
+      return std::to_string(static_cast<uint64_t>(mInteger));
+   case XMLAttributeValueView::Type::Float:
+      return std::to_string(mFloat);
+   case XMLAttributeValueView::Type::Double:
+      return std::to_string(mDouble);
+   case XMLAttributeValueView::Type::StringView:
+      return std::string(mStringView.Data, mStringView.Length);
+   }
+
+   return {};
+}
+
+wxString XMLAttributeValueView::ToWString() const
+{
+   switch (mType)
+   {
+   case XMLAttributeValueView::Type::Null:
+      return {};
+   case XMLAttributeValueView::Type::SignedInteger:
+      return wxString() << mInteger;
+   case XMLAttributeValueView::Type::UnsignedInteger:
+      return wxString() << mInteger; static_cast<uint64_t>(mInteger);
```

Comment:
Unused expression, and incorrect result!

Reply:
Ooops